File: native/dart/packages/blocks_runtime/lib/src/browser_launcher.dart
/// Interface for launching OAuth authorize URLs in a browser.
///
/// Platform implementations (e.g. Flutter) provide concrete launchers
/// that open the system browser and capture the redirect callback.
abstract class BrowserLauncher {
  /// Opens [authorizeUrl] in the system browser and returns the redirect URI
  /// containing the authorization code.
  Future<Uri> launch(Uri authorizeUrl, {required String callbackScheme});
}

File: native/swift/Sources/BlocksRuntime/OIDC/BrowserLauncher.swift
import Foundation

/// Launches an OAuth authorize URL in the system browser and returns the
/// redirect callback URL containing the authorization code.
///
/// On iOS / macOS the typical implementation wraps `ASWebAuthenticationSession`.
/// Apps that drive the browser themselves (e.g. server-rendered hosting) can
/// implement this with their own browser shim.
public protocol BrowserLauncher: Sendable {
    func launch(authorizeURL: URL, callbackScheme: String) async throws -> URL
}
